### ISSUE TITLE
Problem of TimeUpsamle2x, decoder output frames != encoder input frames 

### ISSUE THREAD

**Issue by SamitHuang** [PKU-YuanGroup/Open-Sora-Plan#174]
Due to the first frame is excluded from interpolation in TimeUpsamle2x, as follows in code,
```python
x,x_= x[:,:,:1],x[:,:,1:]
x_= F.interpolate(x_, scale_factor=(2,1,1), mode='trilinear')
x = torch.concat([x, x_], dim=2)
```

decoder output frames != encoder input frames, which looks wrong for reconstruction tasks, e.g. input frames=16, output frames is 13 in running.

Why not directly upsample the whole temporal sequential? 
```
x = F.interpolate(x, scale_factor=(2,1,1), mode='trilinear')
```

Btw, will it be better to use convolution to do the upsampling?

**Comment by SamitHuang** [comment]
"we consider the first frame of a video to be an image..."  I see, the first frame is always encoded from the repeated k-1 1st frames. But for upsampling, the interpolation between the first frame and the second frame doesn't conflict this padding logic. 

Let's say the input of Upsample is z of shape (b c t h w) = (1 512 4 64 64), if we directly upsample the whole temporal sequence from 4 to 8, the output first frame will still correspond to the first image.

**Comment by SamitHuang** [comment]
"we consider the first frame of a video to be an image..."  I see, the first frame is always encoded from the repeated k-1 1st frames. But for upsampling, the interpolation between the first frame and the second frame doesn't conflict this padding logic. 

Let's say the input of Upsample is z of shape (b c t h w) = (1 512 4 64 64), if we directly upsample the whole temporal sequence from 4 to 8, the output first frame will still correspond to the first image.

**Comment by qqingzheng** [comment]
In the downsampling of the encoder, the first frame remains unchanged, so consistency should be maintained in the decoder. Moreover, if the first frame is repeated in the Upsample layer, it will cause problems in subsequent CausalConv3D layers. This is just my understanding, not necessarily reasonable.
